taker-frontend/src/components/Wallet.tsx

```diff
@@ -13,14 +40,54 @@ export default function Wallet(
         walletInfo,
     }: WalletProps,
 ) {
+    const toast = useToast();
     const { hasCopied, onCopy } = useClipboard(walletInfo ? walletInfo.address : "");
     const { balance, address, last_updated_at } = walletInfo || {};
 
+    const [withdrawAmount, setWithdrawAmount] = useState(0);
+    const onWithdrawAmountChange = (amount: string) => {
+        setWithdrawAmount(Number.parseFloat(amount));
+    };
+    const [fee, setFee] = useState(1);
+    const onFeeChange = (amount: string) => {
+        setFee(Number.parseFloat(amount));
+    };
+    const [withdrawAddress, setWithdrawAddress] = useState("");
+    const onWithdrawAddressChange = (address: string) => {
+        setWithdrawAddress(address);
+    };
+
+    let { run: runWithdraw, isLoading: isWithdrawing } = useAsync({
+        deferFn: async ([event]: FormEvent<HTMLFormElement>[]) => {
+            event.preventDefault();
+            try {
+                const url = await postWithdraw({
+                    amount: withdrawAmount,
+                    fee,
+                    address: withdrawAddress,
+                });
+                toast({
+                    title: "Withdraw successful",
+                    description: <Link href={url} isExternal>
+                        {url}
+                    </Link>,
+                    status: "info",
+                    duration: 10000,
+                    isClosable: true,
+                });
```

Comment:
I think I'd prefer if we open a new tab in case the response is a URL. The toast may go away and then the user has no way of retrieving the transaction ID.

Reply:
I think ideally, we would persist these transaction IDs to the DB and display them as a "withdrawal TX list" underneath the wallet.

Reply:
> I think I'd prefer if we open a new tab in case the response is a URL. The toast may go away and then the user has no way of retrieving the transaction ID.

I'll do both. Useful in case a user has blocked the site from opening new tabs.

Reply:
> I think ideally, we would persist these transaction IDs to the DB and display them as a "withdrawal TX list" underneath the wallet.

I think this is a topic story and goes towards showing a complete history of the wallet actions/transactions.